Makefile

```diff
@@ -714,7 +714,7 @@ $(YQ):
 $(GOLANGCI_LINT): .github/workflows/golangci-lint.yml # Download golangci-lint using hack script into tools folder.
 	hack/ensure-golangci-lint.sh \
 		-b $(TOOLS_BIN_DIR) \
-		$(shell cat .github/workflows/golangci-lint.yml | grep [[:space:]]version | sed 's/.*version: //')
+		$(shell yq e '.jobs.golangci.steps[] | select(.name? == "golangci-lint") | .with.version' .github/workflows/golangci-lint.yml )
```

Comment:
Just curious, what does this change do?

Reply:
I briefly explained it in the commit message: https://github.com/kubernetes-sigs/cluster-api-addon-provider-helm/pull/111/commits/13371a063bb71358f5145b85f23176c1cb6a301f
In short, the existing combination of grep+sed doesn't parse .github/workflows/golangci-lint.yml correctly and the script fails. I replaced it with more reliable yq, so we won't have to worry about changes in the config file anymore.